pseudomonas_analysis/2_identify_generic_genes_pathways_pao1.ipynb

```diff
@@ -0,0 +1,1764 @@
+       "KEGG-Module-M00157: F-type ATPase, prokaryotes ...                   0.000194   \n",
+       "\n",
+       "                                                    Number of experiments (simulated)  \\\n",
+       "pathway                                                                                 \n",
+       "KEGG-Module-M00178: Ribosome, bacteria                                             25   \n",
+       "KEGG-Pathway-pae03010: Ribosome - Pseudomonas a...                                 25   \n",
+       "KEGG-Module-M00009: Citrate cycle (TCA cycle, K...                                 25   \n",
+       "KEGG-Module-M00011: Citrate cycle, second carbo...                                 25   \n",
+       "KEGG-Module-M00157: F-type ATPase, prokaryotes ...                                 25   \n",
+       "\n",
+       "                                                     Z score  \n",
+       "pathway                                                       \n",
+       "KEGG-Module-M00178: Ribosome, bacteria             -1.393957  \n",
+       "KEGG-Pathway-pae03010: Ribosome - Pseudomonas a... -1.393957  \n",
+       "KEGG-Module-M00009: Citrate cycle (TCA cycle, K... -1.393957  \n",
+       "KEGG-Module-M00011: Citrate cycle, second carbo... -1.073844  \n",
+       "KEGG-Module-M00157: F-type ATPase, prokaryotes ... -1.394352  "
+      ]
+     },
+     "execution_count": 27,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Create intermediate file: \"<local_dir>/gene_summary_table_<col_to_rank_pathways>.tsv\"\n",
+    "summary_pathway_ranks = ranking.generate_summary_table(\n",
+    "    template_GSEA_stats_filename,\n",
+    "    template_GSEA_stats,\n",
+    "    simulated_GSEA_summary_stats,\n",
+    "    col_to_rank_pathways,\n",
+    "    local_dir,\n",
+    "    'pathway',\n",
+    "    params\n",
+    ")\n",
+    "\n",
+    "summary_pathway_ranks.sort_values(by='Rank (simulated)', ascending=False).head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 28,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Create `pathway_summary_filename`\n",
+    "summary_pathway_ranks.to_csv(pathway_summary_filename, sep='\\t')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Quick check:**\n",
+    "\n",
+    "Looks like Ribosomes are found to be significantly differentiated in the template and across 25 simulated experiments. So this pathway looks to be generic, which is consistent with findings from [Crow. et. al.](https://www.pnas.org/content/116/13/6491.abstract), using Human data. This is also consistent with [paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2679180/) which observed that Ribosome proteins have many different functions in *S. cerevisiae*, which might explain their variability across a wide range of contexts.\n",
```

Comment:
Is there a reason why all the pathways in this table have the same p-value?

Reply:
Good eye. I did check that not all pathways have the same p-value. Looks like the adjusted-pvalue  is a ratio of two distributions: (1) the actual enrichment score versus the enrichment scores for all gene sets against all permutations of the dataset and (2) the actual enrichment score versus the enrichment scores of all gene sets against the actual dataset. Looking at the distribution of FDR there tends to be a pileup at the extremes which I think explains this result.